code/src/nuvla/job/job.py

```diff
@@ -265,11 +265,11 @@ def get_user_api(self):
                                 f'{authn_info["active-claim"]} '
                                 f'{" ".join(authn_info["claims"])}')
 
-    @property
-    def target_resource(self) -> str:
-        return self['target-resource']['href']
+    # @property
+    # def target_resource(self) -> str:
+      #   return self['target-resource']['href']
 
-    nuvlaedge_id = target_resource
+    # nuvlaedge_id = target_resource
```

Comment:
I think you can keep the `target_resource` property, only `nuvlaedge_id` should be removed.

I would maybe rename it `target_resource_href` to be more explicit that what we get is only an href.

Also I would add a docstring mentioning that `target-resource` is not mandatory in the spec so that this function can fail with `KeyError`.